### PR TITLE
Np 10540 referral rewards menu

### PR DESCRIPTION
**Pull request by sarthak-marwaha**
- Add referral border and enhance referral view
- Improve Stack's handling of empty headers

### CODE REVIEW

File: src/foam/nanos/referral/ReferUserView.js [hunk 1]
```diff
@@ -135,3 +196,59 @@ foam.CLASS({
     }
   ]
 });
+
+
+foam.CLASS({
```
Comment (jlhughes) [inline src/foam/nanos/referral/ReferUserView.js]: Please put this in it's own source file.

**Review by jlhughes — APPROVED** [review]

**Comment by adamfox-nanopay** [comment]
This is using hard coded values?

**Comment by sarthak-marwaha** [comment]
> This is using hard coded values?

@adamfox-nanopay for now, joel is working on a way to fetch values on view load, we dont have a dao/service for that

File: src/foam/nanos/referral/ReferUserView.js [hunk 2]
```diff
@@ -55,60 +81,95 @@ foam.CLASS({
   ],
 
    messages: [
-     { name: 'HEADER_1', message: 'Share ' },
-     { name: 'HEADER_2', message: ' With a Friend!' },
+     { name: 'HEADER_1', message: 'Share ${appName} With a Friend!', template: true },
+     { name: 'HEADER_2', message: 'You both get payment credits when your friend completes their first transaction'},
```
Comment (jlhughes) [inline src/foam/nanos/referral/ReferUserView.js:85]: This is nanopay specific.  As well as a couple others

**Review by adamfox-nanopay — APPROVED** [review]

**Review by jlhughes — APPROVED** [review]

File: src/foam/nanos/referral/ReferUserView.js (hunk 2 above)
Comment (sarthak-marwaha) [reply src/foam/nanos/referral/ReferUserView.js:85]: Right, can make this an empty message here and override in locales

**Pull request merged**